docs/en/engines/database-engines/overlay.md

```diff
@@ -0,0 +1,167 @@
+---
+title: 'Overlay'
+slug: /engines/database-engines/overlay
+description: 'Creates a logical read-only database that exposes the union of tables from multiple underlying databases.'
+sidebar_label: 'Overlay'
+sidebar_position: 51
+---
+
+# `DatabaseOverlay` — design & behavior (current implementation)
+
+## What is `DatabaseOverlay`? {#introduction}
+
+`DatabaseOverlay` is a logical “view” database that exposes **the union of tables from multiple underlying databases**. It does **not** own data itself. It supports two modes:
+
+* `Mode::OwnedMembers` — used by **clickhouse-local** helper code; the overlay sits *on top of* member DBs it creates/owns.
```

Comment:
There are no such modes anymore, and maybe we should just describe what are the modes (everying after dash) w/o exposing the internals

Also let's describe the reason for the overlay database in `clickhouse-local`, you can find details here - https://github.com/ClickHouse/ClickHouse/pull/48821